sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala

```diff
@@ -728,9 +728,15 @@ class DataFrameAggregateSuite extends QueryTest with SharedSQLContext {
     import java.lang.Float.floatToRawIntBits
     import java.lang.Double.doubleToRawLongBits
 
-    // 0.0/0.0 and NaN are different values.
-    assert(floatToRawIntBits(0.0f/0.0f) != floatToRawIntBits(Float.NaN))
-    assert(doubleToRawLongBits(0.0/0.0) != doubleToRawLongBits(Double.NaN))
+    if (System.getProperty("os.arch").contains("aarch64")) {
+      // 0.0/0.0 and NaN are same value on aarch64.
+      assert(floatToRawIntBits(0.0f/0.0f) == floatToRawIntBits(Float.NaN))
```

Comment:
We can change it to any value that is NaN but has a different binary representation than `Float.NaN`.